NFIQ2/NFIQ2Algorithm/src/nfiq2/nfiq2impl.cpp

```diff
@@ -154,3 +155,32 @@ NFIQ2Algorithm::Impl::getParameterHash() const
 {
 	return (this->m_parameterHash);
 }
+
+unsigned int
+NFIQ::NFIQ2Algorithm::Impl::computeQualityScore(
+    NFIQ::FingerprintImageData rawImage) const
+{
+
+	std::vector<NFIQ::ActionableQualityFeedback> actionableQuality {};
+	std::vector<NFIQ::QualityFeatureData> qualityFeatureData {};
+	std::vector<NFIQ::QualityFeatureSpeed> qualityFeatureSpeed {};
+
+	return computeQualityScore(rawImage, true, actionableQuality, true,
+	    qualityFeatureData, true, qualityFeatureSpeed);
+}
+
+NFIQ::NFIQ2Results
+NFIQ::NFIQ2Algorithm::Impl::computeQualityFeaturesAndScore(
+    NFIQ::FingerprintImageData rawImage) const
+{
+	std::vector<NFIQ::ActionableQualityFeedback> actionableQuality {};
+	std::vector<NFIQ::QualityFeatureData> qualityfeatureData {};
+	std::vector<NFIQ::QualityFeatureSpeed> qualityFeatureSpeed {};
+
+	const unsigned int qualityScore = computeQualityScore(rawImage, true,
+	    actionableQuality, true, qualityfeatureData, true,
+	    qualityFeatureSpeed);
+
+	return NFIQ::NFIQ2Results(actionableQuality, qualityfeatureData,
+	    qualityFeatureSpeed, qualityScore);
```

Comment:
I think we can avoid the copy of three vectors here. Let's think about this. Can we instantiate the `NFIQ2Results` object first, and have `computeQualityScore` write directly to those vectors?

Reply:
OBE. We will change this very very soon.